movingpandas/trajectory_smoother.py

```diff
@@ -0,0 +1,220 @@
+            return self._smooth_traj_collection(**kwargs)
+        else:
+            raise TypeError
+
+    def _smooth_traj_collection(self, **kwargs):
+        smoothed = []
+        for traj in self.traj:
+            smoothed.append(self._smooth_traj(traj, **kwargs))
+        result = copy(self.traj)
+        result.trajectories = smoothed
+        return result
+
+    @abstractmethod
+    def _smooth_traj(self, traj, **kwargs):
+        raise NotImplementedError
+
+
+class KalmanSmootherCV(TrajectorySmoother):
+    """
+    Smooths using a Kalman Filter with a Constant Velocity model.
+
+    The Constant Velocity model assumes that the speed between consecutive locations is
+    nearly constant. For trajectories where ``traj.is_latlon = True`` the smoother
+    converts to EPSG:3395 (World Mercator) internally to perform filtering and smoothing
+
+    .. note::
+        This class makes use of
+        `Stone Soup <https://stonesoup.readthedocs.io/en/latest/>`_, which is an
+        optional dependency and not installed by default. To use this class, you need
+        to install Stone Soup directly
+        (see `here <https://stonesoup.readthedocs.io/en/latest/#installation>`_).
+    """
+
+    def smooth(self, process_noise_std=0.5, measurement_noise_std=1):
+        """
+        Smooth the input Trajectory/TrajectoryCollection
+
+        Parameters
+        ----------
+        process_noise_std: float or sequence of floats of length 2, default is 1e-5.
+            The process (acceleration) noise standard deviation.
+
+            If a sequence (e.g. list, tuple, etc.) is provided, the first index is used
+            for the x coordinate, while the second is used for the y coordinate. If
+            ``traj.is_latlon=True`` the values are applied to the  easting and northing
+            coordinate (in EPSG:3395) respectively.
+
+            Alternatively, a single float can be provided, which is assumed to be the
+            same for both coordinates.
+        measurement_noise_std: float or sequence of floats of length 2, default is 1.
+            The measurement noise standard deviation.
+
+            If a sequence (e.g. list, tuple, etc.) is provided, the first index is used
+            for the x coordinate, while the second is used for the y coordinate. If
+            ``traj.is_latlon=True`` the values are applied to the  easting and northing
+            coordinate (in EPSG:3395) respectively.
+
+            Alternatively, a single float can be provided, which is assumed to be the
+            same for both coordinates.
+        """
+        return super().smooth(
+            process_noise_std=process_noise_std,
+            measurement_noise_std=measurement_noise_std,
+        )
+
+    def _smooth_traj(self, traj, process_noise_std=0.5, measurement_noise_std=1):
+        # Get detector
+        detector = self._get_detector(traj)
+
+        # Models
+        if not isinstance(process_noise_std, (list, tuple, np.ndarray)):
```

Comment:
There should be a test where the process_noise_std and measurement_noise_std are provided as tuples.

Reply:
Probably with this additional test, we will also reach the target coverage value.

Reply:
I've added a test with tuples with identical values (v1, v1). Of course it would be nice to also test with different values (v1, v2) but it's good enough for merging.